libraries/standard/mqtt/cbmc/include/mqtt_config.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MQTT_CONFIG_H_
+#define MQTT_CONFIG_H_
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Logging related header files are required to be included in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define LIBRARY_LOG_NAME and  LIBRARY_LOG_LEVEL.
+ * 3. Include the header file "logging_stack.h".
+ */
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Configure name and log level for the MQTT library. */
+#define LIBRARY_LOG_NAME     "MQTT"
+#define LIBRARY_LOG_LEVEL    LOG_NONE
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+/* Set network context to a socket (int). */
+typedef int NetworkContext_t;
+
+/**
+ * @brief The maximum number of MQTT PUBLISH messages that may be pending
+ * acknowledgement at any time.
+ *
+ * QoS 1 and 2 MQTT PUBLISHes require acknowledgement from the server before
+ * they can be completed. While they are awaiting the acknowledgement, the
+ * client must maintain information about their state. The value of this
+ * macro sets the limit on how many simultaneous PUBLISH states an MQTT
+ * context maintains.
+ */
+#define MQTT_STATE_ARRAY_MAX_COUNT    ( 10U )
```

Comment:
Is this protocol definition or a bound for the sake of proof?

This file feels like it is mixing MQTT protocol configuration with MQTT proof bounds and stubs.  If that is the case, I would put all of the MQTT stuff in one place (eg the state header).  It is important for a newcomer to be able to go to exactly one place (or two well-documented places) to find all the proof assumptions (eg, queue bounds) and stubs.

Reply:
I added this config header because I needed to disable logging in the library. This symbol is required for MQTT and must be defined. Before we took the definition from the plaintext demo. 
This is a typical count for all of the demos and not for the sake of bounding a proof.

Reply:
I really think a one-line comment saying "10 is a typical value used in demos" would be valuable (we get hammered so hard for bounding lengths, it is relief to be able to say this is not a limitation of the proof).

Reply:
yes thank you